static/src/javascripts/projects/commercial/modules/comscore.js

```diff
@@ -24,47 +25,51 @@ const getGlobals = (consentState, keywords) => {
 	return globals;
 };
 
-const initOnConsent = (state) => {
-	if (!initialised) {
-		// eslint-disable-next-line no-underscore-dangle
-		window._comscore = window._comscore || [];
+const initOnConsent = () => {
+	// eslint-disable-next-line no-underscore-dangle
+	window._comscore = window._comscore || [];
 
-		// eslint-disable-next-line no-underscore-dangle
-		window._comscore.push(
-			getGlobals(!!state, config.get('page.keywords', '')),
-		);
+	// eslint-disable-next-line no-underscore-dangle
+	window._comscore.push(getGlobals(config.get('page.keywords', '')));
 
-		loadScript(comscoreSrc, { id: 'comscore', async: true });
-
-		initialised = true;
-	}
+	return loadScript(comscoreSrc, { id: 'comscore', async: true });
 };
 
-export const init = () => {
+const setupComscore = () => {
 	if (commercialFeatures.comscore) {
-		void getInitialConsentState().then((state) => {
-			/* Rule is that comscore can run:
+		return getInitialConsentState()
+			.then((state) => {
+				/* Rule is that comscore can run:
                 - in Tcfv2: Based on consent for comscore
                 - in Australia: Always
                 - in CCPA: If the user hasn't chosen Do Not Sell
-            */
-			const canRunTcfv2 = state.tcfv2 && getConsentFor('comscore', state);
-			const canRunAus = !!state.aus;
-			const canRunCcpa = !!state.ccpa && !state.ccpa.doNotSell;
+                */
+				const canRunTcfv2 =
+					state.tcfv2 && getConsentFor('comscore', state);
+				const canRunAus = !!state.aus;
+				const canRunCcpa = !!state.ccpa && !state.ccpa.doNotSell;
 
-			if (canRunTcfv2 || canRunAus || canRunCcpa) initOnConsent(true);
-		});
+				if (!(canRunTcfv2 || canRunAus || canRunCcpa)) {
+					return Promise.reject('No consent for comscore');
+				}
+				return initOnConsent();
+			})
+			.catch((e) => {
+				log('commercial', '⚠️ Failed to execute comscore', e);
+			});
 	}
-
 	return Promise.resolve();
 };
 
+const setupComscoreOnce = once(setupComscore);
```

Comment:
Using `once` rather than a `initialised` / `resetInit` in line with the other commercial modules we only want to initialise once.

Reply:
I think this was only done so we could test the internals better in in 87e37c0dcf197c377dffa51dc444a20ef8db5343. Just a FYI, no action needed.

I think it might be worth reviewing how we handle these dependencies in the future, and maybe move away from Lodash’s `once`?